packages/cells/src/model.ts

```diff
@@ -326,20 +356,28 @@ export class CellModel extends CodeEditor.Model implements ICellModel {
    */
   protected _onSharedModelChanged(
     sender: nbmodel.ISharedCodeCell,
-    change: nbmodel.CellChange<nbformat.ICodeCellMetadata>
+    change: nbmodel.CellChange<nbmodel.ISharedBaseCellMetada>
   ): void {
     super._onSharedModelChanged(sender, change);
     this._mutex(() => {
       if (change.metadataChange) {
-        const newValue = change.metadataChange?.newValue;
-        if (newValue?.collapsed)
-          this.metadata.set('collapsed', newValue?.collapsed);
-        if (newValue?.jupyter) this.metadata.set('jupyter', newValue?.jupyter);
-        if (newValue?.name) this.metadata.set('name', newValue?.name);
-        if (newValue?.scrolled)
-          this.metadata.set('scrolled', newValue?.scrolled);
-        if (newValue?.tags) this.metadata.set('tags', newValue?.tags);
-        if (newValue?.trusted) this.metadata.set('trusted', newValue?.trusted);
+        const newValue = change.metadataChange
+          ?.newValue as nbmodel.ISharedBaseCellMetada;
+        if (newValue) {
+          if (newValue.collapsed) {
+            if (!this.metadata.get('jupyter')) this.metadata.set('jupyter', {});
+            (this.metadata.get('jupyter') as any)!['outputs_hidden'] =
+              newValue.collapsed;
+          }
+          if (newValue.jupyter) this.metadata.set('jupyter', newValue.jupyter);
+          if (newValue.name) this.metadata.set('name', newValue.name);
+          if (newValue.scrolled)
+            this.metadata.set('scrolled', newValue.scrolled);
+          if (newValue.tags) this.metadata.set('tags', newValue.tags);
+          if (newValue.trusted) this.metadata.set('trusted', newValue.trusted);
```

Comment:
We should also check if the specific value actually changed. Otherwise, we would set six values in modeldb every time any value in nmbodel.metadata is updated.

Reply:
this case is handled by the set method withing modelDB. IFRM it directly returns if the value does not change

```
  set(key: string, value: T): T | undefined {
    const oldVal = this._map.get(key);
    if (value === undefined) {
      throw Error('Cannot set an undefined value, use remove');
    }
    // Bail if the value does not change.
    const itemCmp = this._itemCmp;
    if (oldVal !== undefined && itemCmp(oldVal, value)) {
      return oldVal;
    }
```

Reply:
Oh perfect!